### PR TITLE
[Serializer] Unified normalizers/encoders config through default context solely

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #28709   
| License       | MIT
| Doc PR        | N/A

As planned in #28709.
Split into two commits to ease review.

### CODE REVIEW

File: src/Symfony/Component/Serializer/Tests/Normalizer/ObjectNormalizerTest.php [hunk 1]
```diff
@@ -31,10 +30,8 @@
 use Symfony\Component\Serializer\SerializerInterface;
 use Symfony\Component\Serializer\Tests\Fixtures\CircularReferenceDummy;
 use Symfony\Component\Serializer\Tests\Fixtures\GroupDummy;
-use Symfony\Component\Serializer\Tests\Fixtures\MaxDepthDummy;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/ObjectNormalizerTest.php]: should the fixture file be removed? if yes what about other that may now be unused?

Comment (ogizanagi) [reply src/Symfony/Component/Serializer/Tests/Normalizer/ObjectNormalizerTest.php]: Well, this one is still used by many tests (see `\Symfony\Component\Serializer\Tests\Normalizer\Features\MaxDepthTestTrait::testMaxDepthHandler`) for instance.
But I'll have a closer look at it soon :)

**Comment by ogizanagi** [comment]
Rebased & green

File: src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php [hunk 2]
```diff
@@ -93,7 +91,7 @@ public function testNormalize()
             ->expects($this->once())
             ->method('normalize')
             ->with($object, 'any')
-            ->will($this->returnValue('string_object'))
+            ->willReturn('string_object')
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php]: I don't know why fabbot is adding these now, but this should be done on lower branches first and reverted here

**Comment by nicolas-grekas** [comment]
Thank you @ogizanagi.

File: src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Serializer/Tests/Normalizer/GetSetMethodNormalizerTest.php]: reverted in 0dd0679362 according to #31737

**Pull request merged**